File: Part 2 - Regression/Section 6 - Polynomial Regression/polynomial_regression.py
# Data Preprocessing Template

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2:3].values

# Fitting linear
from sklearn.linear_model import LinearRegression
lin_reg = LinearRegression()
lin_reg.fit(X, y)

# Fitting polynomial
from sklearn.preprocessing import PolynomialFeatures
poly_reg = PolynomialFeatures(degree = 4)
X_poly = poly_reg.fit_transform(X)
lin_reg_poly = LinearRegression()
lin_reg_poly.fit(X_poly, y)

# Visualize linear
plt.scatter(X, y, color = 'red')
plt.plot(X, lin_reg.predict(X), color = 'blue')
plt.title('Truth or Bluff (Linear)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()

# Visualize poly
X_grid = np.arange(min(X), max(X), 0.1)
X_grid = X_grid.reshape((len(X_grid), 1))
plt.scatter(X, y, color = 'red')
plt.plot(X_grid, lin_reg_poly.predict(poly_reg.fit_transform(X_grid)), color = 'blue')
plt.title('Truth or Bluff (Poly)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()

# Predict with linear
lin_reg.predict(6.5)

# Predict with poly
lin_reg_poly.predict(poly_reg.fit_transform(6.5))

File: Part 2 - Regression/Section 6 - Polynomial Regression/polynomial_regression.R
# Data Preprocessing Template

# Importing the dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

# Fit Linear
lin_reg = lm(formula = Salary ~ ., data = dataset)

# Fit Polynomial
dataset$Level2 = dataset$Level ^ 2
dataset$Level3 = dataset$Level ^ 3
dataset$Level4 = dataset$Level ^ 4
poly_reg = lm(formula = Salary ~ ., data = dataset)

# Visualize lin
ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary), color = 'red') +
  geom_line(aes(x = dataset$Level, y = predict(lin_reg, newdata = dataset)), color = 'blue') +
  ggtitle('Truth or Bluff (Linear)') +
  xlab('Level') +
  ylab('Salary')

# Visualize poly
ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary), color = 'red') +
  geom_line(aes(x = dataset$Level, y = predict(poly_reg, newdata = dataset)), color = 'blue') +
  ggtitle('Truth or Bluff (Polynomial)') +
  xlab('Level') +
  ylab('Salary')

# Predict lin
y_pred = predict(lin_reg, data.frame(Level = 6.5))

# Predict poly
y_poly_pred = predict(poly_reg, data.frame(Level = 6.5, Level2 = 6.5^2, Level3 = 6.5^3, Level4 = 6.5^4))
